Document mixed-mode authentication and authorization (#1126)
neo4j/docs-operations: modules/ROOT/pages/authentication-authorization/index.adoc

It is recommended that this option is used as part of a custom delivery as negotiated with link:https://neo4j.com/professional-services/[Neo4j Professional Services].
For more information, see link:{neo4j-docs-base-uri}/java-reference/{page-version}/extending-neo4j/security-plugins#extending-neo4j-security-plugins[Java Reference -> Authentication and authorization plugins].

*Kerberos authentication and single sign-on*::
In addition to LDAP, native, and custom providers, Neo4j supports Kerberos for authentication and single sign-on.
Kerberos support is provided via the link:{neo4j-docs-base-uri}/kerberos-add-on/current/[Neo4j Kerberos Add-On].

> *Mixed-mode authentication

Author: ```suggestion
*Mixed-mode authentication*::
```

Neo4j also supports mixed-mode authentication that allows you to use multiple authentication providers in your database setup.

> For more information and examples, see xref:authentication-authorization/ldap-integration.adoc#auth-ldap-configure-provider[Set Neo4j to use LDAP] and xref:authentication-authorization/sso-integration.adoc#auth-sso-configure-sso[Configure Neo4j to use OpenID Connect].

Author: so we don't need here referenced to the configuration settings, do we?

Reviewer: I don't think so. Here is just a list of all supported providers with links to more information. We don't need to explain how to configure them. If we decide to do that, we need to rewrite the page and add that information for all of them.

It is recommended that this option is used as part of a custom delivery as negotiated with link:https://neo4j.com/professional-services/[Neo4j Professional Services].
For more information, see link:{neo4j-docs-base-uri}/java-reference/{page-version}/extending-neo4j/security-plugins#extending-neo4j-security-plugins[Java Reference -> Authentication and authorization plugins].

*Kerberos authentication and single sign-on*::
In addition to LDAP, native, and custom providers, Neo4j supports Kerberos for authentication and single sign-on.
Kerberos support is provided via the link:{neo4j-docs-base-uri}/kerberos-add-on/current/[Neo4j Kerberos Add-On].

> 

Reviewer: ```suggestion
*Mixed-mode authentication
```

> In addition, Neo4j supports mixed-mode authentication and authorization that allows to use multiple authentication providers in your database setup.

Reviewer: I am not sure if we need all this information here on the index page where we just list the providers. I think just something like this would suffice:
```suggestion
Neo4j also supports mixed-mode authentication that allows you to use multiple authentication providers in your database setup.
For more information and examples, see xref:authentication-authorization/ldap-integration.adoc#auth-ldap-configure-provider[Set Neo4j to use LDAP] and xref:authentication-authorization/sso-integration.adoc#auth-sso-configure-sso[Configure Neo4j to use OpenID Connect].
```

It is recommended that this option is used as part of a custom delivery as negotiated with link:https://neo4j.com/professional-services/[Neo4j Professional Services].
For more information, see link:{neo4j-docs-base-uri}/java-reference/{page-version}/extending-neo4j/security-plugins#extending-neo4j-security-plugins[Java Reference -> Authentication and authorization plugins].

*Kerberos authentication and single sign-on*::
In addition to LDAP, native, and custom providers, Neo4j supports Kerberos for authentication and single sign-on.
Kerberos support is provided via the link:{neo4j-docs-base-uri}/kerberos-add-on/current/[Neo4j Kerberos Add-On].

*Mixed-mode authentication
Neo4j also supports mixed-mode authentication that allows you to use multiple authentication providers in your database setup.
For more information and examples, see xref:authentication-authorization/ldap-integration.adoc#auth-ldap-configure-provider[Set Neo4j to use LDAP] and xref:authentication-authorization/sso-integration.adoc#auth-sso-configure-sso[Configure Neo4j to use OpenID Connect].

> To configure mixed-mode authentication and authorization in Neo4j, modify the following configuration settings in the _neo4j.conf_ file:

Reviewer: ```suggestion
```

Reviewer: I meant to remove all this information here and be consistent with the rest of the providers.

> * xref:configuration/configuration-settings.adoc#config_dbms.security.authentication_providers[`dbms.security.authentication_providers`]

Reviewer: ```suggestion
```

> * xref:configuration/configuration-settings.adoc#config_dbms.security.authorization_providers[`dbms.security.authorization_providers`]

Reviewer: ```suggestion
```

> These configuration values are comma-separated lists and can include Neo4j native, LDAP, and SSO providers.

Reviewer: ```suggestion
```

> Examples are provided on the following pages:

Reviewer: ```suggestion
```

> * xref:authentication-authorization/ldap-integration.adoc#auth-ldap-configure-provider[Set Neo4j to use LDAP]

Reviewer: ```suggestion
```

> * xref:authentication-authorization/sso-integration.adoc#auth-sso-configure-sso[Configure Neo4j to use OpenID Connect]

Reviewer: ```suggestion
```
